tests/demo/main.c

```diff
@@ -1,35 +1,36 @@
-#include <stdio.h>
 #include <stdlib.h>
-
-
-#define PIO_BASE 0x60000
+#include <stdint.h>
+#include "pio_drv_simple.h"
+#include "metal_pio.h"
 
 int main()
 {
   // read/write to axi block
-  volatile uint32_t * pio = (uint32_t *) PIO_BASE;
-
-  volatile uint32_t * odata   = pio;
-  volatile uint32_t * oenable = pio + 1;
-  volatile uint32_t * idata   = pio + 2;
-
   uint32_t odatas[5] = {0xDEADBEEF, 0xF0F0F0F0, 0xABCD1234, 0x01234567, 0xFEDCBA98};
   uint32_t oenables[5] = {0xF0F0F0F0, 0x0F0F0F0F, 0xDEADBEEF, 0x89ABCDEF, 0x7654321};
+  uint32_t read_value;
+  const struct metal_pio * m_pio = get_metal_pio();
```

Comment:
How will this handle the case where there are multiple PIO instances in a design?

Reply:
will redo